shell/platform/android/io/flutter/plugin/common/MethodChannel.java

```diff
@@ -152,6 +156,54 @@ public void setMethodCallHandler(final MethodCallHandler handler) {
         void notImplemented();
     }
 
+    /**
+     * An exception that {@link MethodCallHandler} implementations may throw in
+     * {@link MethodCallHandler#onMethodCall(MethodCall, Result)} that will be caught by the
+     * channel implementation and sent back to Flutter as a {@code PlatformException}. Metadata
+     * in this exception class will be translated to corresponding fields in the Dart exception.
+     *
+     * <p>Other uncaught exceptions thrown by {@code MethodCallHandler} implementations will also
+     * be sent back to Flutter as a {@code PlatformException}, but they will have a generic
+     * "error" code and no error details.</p>
+     */
+    public static final class ResultException extends RuntimeException {
```

Comment:
Bummer. Having multiple ways to do the same thing is not a good sign for an API.

Would it be worth having a distinction between async and sync use cases?

- T onMethodCall(Params) throws ResultException {}

vs

- void onMethodCallAsync(Params, FlutterResult) {}

I vaguely recall we had this before.

If we want to collapse these and always have FlutterResult communicate the results, then defining this extra exception class is probably not a good thing. It would confuse devs.